web/src/components/TaskHeader.test.tsx

```diff
@@ -192,6 +212,237 @@ describe("<TaskHeader />", () => {
     });
   });
 
+  describe("tax registration modal", () => {
+    it("opens when tax task is marked complete", () => {
+      const taxTask = generateTask({ id: taxTaskId });
+      const userData = generateUserData({
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+      });
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+      expect(screen.getByText(Config.taxRegistrationModal.title)).toBeInTheDocument();
+    });
+
+    it("doesn't update the task status when closed", async () => {
+      const taxTask = generateTask({ id: taxTaskId });
+      const userData = generateUserData({
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+      });
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+
+      fireEvent.click(screen.getByText(Config.taxRegistrationModal.cancelButtonText));
+      await waitFor(() => {
+        const dropdown = screen.getByTestId("taskProgress");
+        expect(within(dropdown).getByText(Config.taskProgress.NOT_STARTED)).toBeInTheDocument();
+      });
+      expect(userDataWasNotUpdated).toBeTruthy();
+    });
+
+    it("updates the task progress when the modal is saved", async () => {
+      const taxTask = generateTask({ id: taxTaskId });
+      const userData = generateUserData({
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+      });
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+
+      fireEvent.click(screen.getByText(Config.taxRegistrationModal.saveButtonText));
+      await waitFor(() => {
+        const dropdown = screen.getByTestId("taskProgress");
+        expect(within(dropdown).getByText(Config.taskProgress.COMPLETED)).toBeInTheDocument();
+      });
+    });
+
+    it("gets redirected to roadmap with the proper query parameter when completed", async () => {
+      const taxTask = generateTask({ id: taxTaskId });
+      const userData = generateUserData({
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+        profileData: generateProfileData({ businessPersona: "STARTING" }),
+      });
+
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+
+      fireEvent.click(screen.getByText(Config.taxRegistrationModal.saveButtonText));
+      await waitFor(() => {
+        expect(mockPush).toHaveBeenCalledWith({
+          pathname: routeForPersona(userData.profileData.businessPersona),
+          query: { fromFormBusinessEntity: "false", fromTaxRegistration: "true" },
+        });
+      });
+    });
+
+    it("shows the warning modal if the user tries to change status from completed", () => {
+      const taxTask = generateTask({ id: taxTaskId });
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          businessName: "test-business",
+          legalStructureId:
+            randomInt() % 2 ? randomTradeNameLegalStructure() : randomPublicFilingLegalStructure(),
```

Comment:
in the spirit of simplifying tests, if this could be either legal structure, can we just get rid of this line and let the generator pick one randomly? And same for businessName, can this be deleted here?

Reply:
same comment in test below